src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -213,6 +219,25 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		return tabsScrollbar;
 	}
 
+	private setVisibility(visible: boolean): void {
+		this.visible = visible;
+		const tabsAndActionsContainer = assertIsDefined(this.tabsAndActionsContainer);
+		tabsAndActionsContainer.style.display = visible ? 'flex' : 'none';
+	}
+
+	override showEditorActionsToolbar(): void {
+		this.setEditorActionsToolbarVisibility(true);
+	}
+
+	override hideEditorActionsToolbar(): void {
+		this.setEditorActionsToolbarVisibility(false);
+	}
+
+	private setEditorActionsToolbarVisibility(visible: boolean): void {
+		const editorToolbarContainer = assertIsDefined(this.editorToolbarContainer);
+		editorToolbarContainer.style.display = visible ? 'flex' : 'none';
+	}
```

Comment:
Can we try to do this with CSS only? Both visibility of tabs bar and actions container seems to be defined as `display: flex` and `display: none`. 

For tabs bar visibility, we could toggle a classname `empty` on the tabs container to drive this.

For actions bar visibility, we could have a classname on the title container for when 2 tab rows are visible, to then disable visibility in the tab bar that is for pinned tabs?

I think we could then remove these methods from the API, esp. since they are only used in the multi-tab control and not the single-tab control or abstract tab control.